server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1326,10 +1327,9 @@ private HypervisorType getHypervisorType(VMInstanceVO vm, StoragePool srcVolumeP
             if (s_logger.isDebugEnabled()) {
                 s_logger.debug("Searching for all hosts in cluster " + cluster + " for migrating VM " + vm);
             }
-            allHostsPair = searchForServers(startIndex, pageSize, null, hostType, null, null, null, cluster, null, keyword, null, null, null, null);
-            // Filter out the current host.
+            allHostsPair = searchForServers(startIndex, pageSize, null, hostType, null, null, null, cluster, null, keyword, null, null, null,
+                null, srcHost.getId());
             allHosts = allHostsPair.first();
-            allHosts.remove(srcHost);
```

Comment:
@davidjumani didn't see any change with respect to hosts count, does these changes fix the count issue?

Reply:
It does since I've added an excludes param in the `searchForServers` method and passing the source host ID to it. This excludes the source host in the query, as well as ensures that the allHosts list is never altered. So the count remains consistant but unsuitable hosts are marked accordingly